samples/DirectX/D3D12/HelloConstBuffer12.cs

```diff
@@ -0,0 +1,807 @@
+                    }
+
+                    var ranges = new D3D12_DESCRIPTOR_RANGE1[1];
+                    const int RootParametersCount = 1;
+                    var rootParameters = new D3D12_ROOT_PARAMETER1[RootParametersCount];
+
+                    ranges[0].Init(D3D12_DESCRIPTOR_RANGE_TYPE.D3D12_DESCRIPTOR_RANGE_TYPE_CBV, 1, 0, 0, D3D12_DESCRIPTOR_RANGE_FLAGS.D3D12_DESCRIPTOR_RANGE_FLAG_DATA_STATIC);
+                    fixed (D3D12_DESCRIPTOR_RANGE1* pRanges = &ranges[0])
+                    {
+                        rootParameters[0].InitAsDescriptorTable(1, pRanges, D3D12_SHADER_VISIBILITY.D3D12_SHADER_VISIBILITY_VERTEX);
+                    }
+
+                    // Allow input layout and deny uneccessary access to certain pipeline stages.
+                    D3D12_ROOT_SIGNATURE_FLAGS rootSignatureFlags =
+                        D3D12_ROOT_SIGNATURE_FLAG_ALLOW_INPUT_ASSEMBLER_INPUT_LAYOUT |
+                        D3D12_ROOT_SIGNATURE_FLAG_DENY_HULL_SHADER_ROOT_ACCESS |
+                        D3D12_ROOT_SIGNATURE_FLAG_DENY_DOMAIN_SHADER_ROOT_ACCESS |
+                        D3D12_ROOT_SIGNATURE_FLAG_DENY_GEOMETRY_SHADER_ROOT_ACCESS |
+                        D3D12_ROOT_SIGNATURE_FLAG_DENY_PIXEL_SHADER_ROOT_ACCESS;
+
+                    var rootSignatureDesc = new D3D12_VERSIONED_ROOT_SIGNATURE_DESC();
+                    fixed (D3D12_ROOT_PARAMETER1* pRootParameters = &rootParameters[0])
+                    {
+                        rootSignatureDesc.Init_1_1(RootParametersCount, pRootParameters, 0, null, rootSignatureFlags);
+                    }
+
+                    ThrowIfFailed(nameof(D3D12SerializeRootSignature), D3D12SerializeVersionedRootSignature(&rootSignatureDesc, featureData.HighestVersion, &signature, &error));
+                    {
+                        iid = IID_ID3D12RootSignature;
+                        ThrowIfFailed(nameof(ID3D12Device.CreateRootSignature), _device->CreateRootSignature(0, signature->GetBufferPointer(), signature->GetBufferSize(), &iid, (void**)&rootSignatureDesc)); 
+                    }
+                }
+
+                // Create an empty root signature.
+                {
+                    var rootSignatureDesc = new D3D12_ROOT_SIGNATURE_DESC {
+                        Flags = D3D12_ROOT_SIGNATURE_FLAG_ALLOW_INPUT_ASSEMBLER_INPUT_LAYOUT
+                    };
+
+                    ThrowIfFailed(nameof(D3D12SerializeRootSignature), D3D12SerializeRootSignature(&rootSignatureDesc, D3D_ROOT_SIGNATURE_VERSION_1, &signature, &error));
+
+                    fixed (ID3D12RootSignature** rootSignature = &_rootSignature)
+                    {
+                        iid = IID_ID3D12RootSignature;
+                        ThrowIfFailed(nameof(ID3D12Device._CreateRootSignature), _device->CreateRootSignature(nodeMask: 0, signature->GetBufferPointer(), signature->GetBufferSize(), &iid, (void**)rootSignature));
+                    }
+                }
+
```

Comment:
This is replaced by the immediately preceding section which creates a proper root signature accounting for the constant buffer
```suggestion
```